app/components/MintNBTC.tsx

```diff
@@ -51,31 +63,36 @@ function Fee({ fee, youReceive }: FeeProps) {
 				</div>
 				<div className="flex justify-between">
 					<p className="text-gray-400">You Receive</p>
-					<p>{youReceive} nBTC</p>
+					<NumericFormat displayType="text" value={youReceive} suffix=" nBTC" />
 				</div>
 			</CardContent>
 		</Card>
 	);
 }
 
 interface MintNBTCForm {
-	numberOfNBTC: string;
+	numberOfBTC: string;
 	suiAddress: string;
 }
 
 export function MintBTC() {
-	const availableBalance = 0;
-	const suiAddress = "";
+	const { balance } = useXverseWallet();
+	const { connectedWallet } = useContext(WalletContext);
+	const isBitCoinWalletConnected = connectedWallet === ByieldWallet.Xverse;
 	const mintNBTCForm = useForm<MintNBTCForm>({
+		mode: "all",
+		reValidateMode: "onChange",
 		defaultValues: {
-			suiAddress,
+			numberOfBTC: "",
+			suiAddress: "",
 		},
 	});
-	const { handleSubmit, watch } = mintNBTCForm;
-	const numberOfBTC = watch("numberOfNBTC");
+	const { handleSubmit, watch, setValue } = mintNBTCForm;
+	const numberOfBTC = watch("numberOfBTC");
 
 	// satoshi. 1BTC = 10^8 satoshi
 	const fee = 0.00000001;
+	const youReceive = Number(numberOfBTC) - fee;
```

Comment:
match should be done using BigInt. So convert everything to BigInt and do operations on BigInt:

```suggestion
	const BTC_FACTOR = BigInt(10**8);  // TODO: put this in denoms.ts module
	const fee = 10n;
	try {
	   const youReceive = BigInt(numberOfBTC) - fee;
	} catch {
       // handle 	numberOfBTC is not a number
    }
    if (youReceive <= 0) { handle "not enough BTC to send to cover fee" }
```